projects/ReinforcementLearning/lunar_lander.ipynb

```diff
@@ -771,36 +778,176 @@
    },
    "outputs": [],
    "source": [
-    "def step(self, actions):\n",
-    "  ...\n",
-    "  ...\n",
-    "  ...\n",
-    "  reward = 0\n",
-    "  shaping = (\n",
-    "      -100 * np.sqrt(state[0] * state[0] + state[1] * state[1])\n",
-    "      - 100 * np.sqrt(state[2] * state[2] + state[3] * state[3])\n",
-    "      - 100 * abs(state[4])\n",
-    "      + 10 * state[6]\n",
-    "      + 10 * state[7]\n",
-    "  )  # And ten points for legs contact, the idea is if you\n",
-    "  # lose contact again after landing, you get negative reward\n",
-    "  if self.prev_shaping is not None:\n",
-    "      reward = shaping - self.prev_shaping\n",
-    "  self.prev_shaping = shaping\n",
-    "\n",
-    "  reward -= (\n",
-    "      m_power * 0.30\n",
-    "  )  # less fuel spent is better, about -30 for heuristic landing. You should modify these values.\n",
-    "  reward -= s_power * 0.03\n",
-    "\n",
-    "  done = False\n",
-    "  if self.game_over or abs(state[0]) >= 1.0:\n",
-    "      done = True\n",
-    "      reward = -100\n",
-    "  if not self.lander.awake:\n",
-    "      done = True\n",
-    "      reward = +100\n",
-    "  return np.array(state, dtype=np.float32), reward, done, {}"
+    "class Custom_LunarLander(LunarLander):\n",
+    "\n",
+    "    def step(self, action):\n",
+    "        assert self.lander is not None\n",
+    "\n",
+    "        # Update wind\n",
+    "        assert self.lander is not None, \"You forgot to call reset()\"\n",
+    "        if self.enable_wind and not (\n",
+    "            self.legs[0].ground_contact or self.legs[1].ground_contact\n",
+    "        ):\n",
+    "            # the function used for wind is tanh(sin(2 k x) + sin(pi k x)),\n",
+    "            # which is proven to never be periodic, k = 0.01\n",
+    "            wind_mag = (\n",
+    "                math.tanh(\n",
+    "                    math.sin(0.02 * self.wind_idx)\n",
+    "                    + (math.sin(math.pi * 0.01 * self.wind_idx))\n",
+    "                )\n",
+    "                * self.wind_power\n",
+    "            )\n",
+    "            self.wind_idx += 1\n",
+    "            self.lander.ApplyForceToCenter(\n",
```

Comment:
did you implement this for them?

Reply:
No, the lines 793-801 are the original ones.